import{_ as e,o as t,c as a,Q as o}from"./chunks/framework.9ad6f510.js";const f=JSON.parse('{"title":"Data Requirements","description":"","frontmatter":{},"headers":[],"relativePath":"preparing-data/data-requirements/index.md","filePath":"preparing-data/data-requirements/index.md"}'),i={name:"preparing-data/data-requirements/index.md"},n=o('<h1 id="data-requirements" tabindex="-1">Data Requirements <a class="header-anchor" href="#data-requirements" aria-label="Permalink to &quot;Data Requirements&quot;">​</a></h1><p>There are three main types of data that you&#39;ll need to use <strong><code>dms-viz</code></strong>; your <a href="#input-data">input data</a>, a <a href="#sitemap">map of the sites</a> in your experiment to a reference and/or protein structure, and <em>optionally</em>, some <a href="#join-data">additional data</a> that you want to join with your input data. Below are details of what this data should look like along with examples.</p><h2 id="input-data" tabindex="-1">Input Data <a class="header-anchor" href="#input-data" aria-label="Permalink to &quot;Input Data&quot;">​</a></h2><p>The <strong>Input Data</strong> is the data that you&#39;d like to summarize and visualize on a protein structure. It must contain a column with a quantitative metric that&#39;s associated with substitutions in a protein sequence. For example, this data could be a fitness score associated with mutations to a viral protein, or a score that represents how a mutation abrogates antibody binding to a protein. For detailed examples of <strong><code>dms-viz</code></strong> being used to visualize this kind of data, check out the <a href="/dms-viz-docs/visualizing-data/vignettes/">Vignettes</a>.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The input data must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><p>The input data must contain the following columns with <strong>exactly</strong> these names:</p><ul><li><h3 id="site-or-reference-site" tabindex="-1"><code>site</code> or <code>reference_site</code> <a class="header-anchor" href="#site-or-reference-site" aria-label="Permalink to &quot;`site` or `reference_site`&quot;">​</a></h3><p>This column should contain the <strong>site</strong> in the protein at which each measurement was made. This column can be numeric (i.e., <code>[1, 2, 3, 4]</code>) or it can contain strings (i.e., <code>[1, 2, 2a, 2b, 3]</code>). Additionally, the sites do not need to be continuous (i.e., <code>[1, 4, 5, 8]</code>). The numeric order of your sites will be specified in the <a href="#sitemap">Sitemap</a>. The sites in this column will be displayed on the x-axis of summary plots.</p><p>For more detail on what we mean by &#39;reference_site&#39;, check out the <a href="#reference-site">description in the sitemap</a>.</p></li><li><h3 id="mutant" tabindex="-1"><code>mutant</code> <a class="header-anchor" href="#mutant" aria-label="Permalink to &quot;`mutant`&quot;">​</a></h3><p>This column should contain the identity of the <strong>mutation</strong> that each measurement is associated with. These mutations should be represented using the <a href="https://www.bioinformatics.org/sms/iupac.html" target="_blank" rel="noreferrer">IUPAC single letter codes</a> along with symbols for stop codons and gaps (i.e., <code>R, M, P, *, -</code>). If you need to extend this alphabet with other symbols, you can do so using the <a href="/dms-viz-docs/preparing-data/command-line-api/#alphabet"><code>--alphabet</code></a> flag.</p></li><li><h3 id="wildtype" tabindex="-1"><code>wildtype</code> <a class="header-anchor" href="#wildtype" aria-label="Permalink to &quot;`wildtype`&quot;">​</a></h3><p>This column should contain the <strong>wildtype</strong> identity of residues at a given site in the protien. For example, if a Proline (<code>P</code>) was mutatated to an Alanine (<code>A</code>) at position 120 in the protein (<code>P120A</code>), there should be a <code>P</code> in the wildtype column for every row where the value of the site column is 120.</p></li></ul><hr><p>In addition to these three mandatory columns, you will also need to specify a <code>metric</code> column. The identity of this column is specified with <a href="/dms-viz-docs/preparing-data/command-line-api/#metric"><code>--metric</code></a> flag and it can have any name:</p><ul><li><h3 id="metric" tabindex="-1"><code>&lt;metric&gt;</code> <a class="header-anchor" href="#metric" aria-label="Permalink to &quot;`&lt;metric&gt;`&quot;">​</a></h3><p>This column should contain the <strong>quantitative metric</strong> that you&#39;d like to summarize and view on a protein structure. For example, this column could be called <code>fitness</code> and contain a score that reflects how individual mutations alter a protein&#39;s function.</p></li></ul><hr><p><em>Optionally</em>, depending on the structure of your experiment, you can also include a &quot;<em>condition column</em>&quot; that specifies how your data is grouped if there are multiple measurements per mutation:</p><ul><li><h3 id="condition" tabindex="-1"><code>condition</code> <a class="header-anchor" href="#condition" aria-label="Permalink to &quot;`condition`&quot;">​</a></h3><p>This column should only be included if there are multiple measurements in the <a href="/dms-viz-docs/preparing-data/command-line-api/#metric"><code>&lt;metric&gt;</code></a> column for the same <code>site</code>/<code>mutation</code> combinations. An example of this would be if you have a measurement like antibody escape for multiple &#39;epitopes&#39; in a protein. This column contains a unique identifier that&#39;s used to deliniate between these measurements for each mutation.</p></li></ul><h2 id="sitemap" tabindex="-1">Sitemap <a class="header-anchor" href="#sitemap" aria-label="Permalink to &quot;Sitemap&quot;">​</a></h2><p>The <strong>Sitemap</strong> is a tabular dataframe that specifies the order of the <a href="#site-or-reference-site"><code>site</code></a> column in your input data and, <em>optionally</em>, how the <a href="#site-or-reference-site"><code>site</code></a> column corresponds to the site numbering in the <a href="/dms-viz-docs/preparing-data/command-line-api/#structure">protein structure</a> you provide.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The sitemap must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><ul><li><h3 id="reference-site" tabindex="-1"><code>reference_site</code> <a class="header-anchor" href="#reference-site" aria-label="Permalink to &quot;`reference_site`&quot;">​</a></h3><p>This column <strong>must</strong> correspond to the <code>site</code> or <code>reference_site</code> column in your <a href="#input-data">input data</a>. If the <a href="#protein-site"><code>protein_site</code></a> isn&#39;t provided, this column is also assumed to correspond to the identity of the sites in the <a href="/dms-viz-docs/preparing-data/command-line-api/#structure">protein structure</a></p><p>The <code>reference_site</code> refers to the labels that are applied to the sites on the x-axis of the plots. These can sometimes differ from the <code>sequential_site</code> (<a href="#sequential-site">described below</a>); for example, the current SARS-CoV-2 Spike protein variants now have indels that cause the widely used Wuhan-Hu-1 &#39;reference&#39; numbering to differ from the sequential, numeric order of the data.</p></li><li><h3 id="sequential-site" tabindex="-1"><code>sequential_site</code> <a class="header-anchor" href="#sequential-site" aria-label="Permalink to &quot;`sequential_site`&quot;">​</a></h3><p>This column is the sequential order of the reference sites and <strong>must be a numeric column</strong>. This will determine the order of the protein sites in the visualizations.</p></li><li><h3 id="protein-site" tabindex="-1"><code>protein_site</code> <a class="header-anchor" href="#protein-site" aria-label="Permalink to &quot;`protein_site`&quot;">​</a></h3><p><em>Optionally</em>, this column is only necessary if the <code>reference_site</code> sites are different from the sites in your provided protein strucutre. If they are different, this column is the position in the protein structure that corresponds to the <code>reference_site</code> values in your data.</p></li></ul><h2 id="join-data" tabindex="-1">Join Data <a class="header-anchor" href="#join-data" aria-label="Permalink to &quot;Join Data&quot;">​</a></h2><p><em>Optionally</em>, you might have some more data that you want to combine with your <a href="#input-data">Input Data</a>. You might want to do this so that you can include columnns from this <strong>Join Data</strong> in the <a href="/dms-viz-docs/preparing-data/command-line-api/#filter-cols">filters</a> or <a href="/dms-viz-docs/preparing-data/command-line-api/#tooltip-cols">tooltips</a> of your visualization. This option is to streamline that workflow.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The join data must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><p>You can specify more than one <code>.csv</code> file if there are multiple sources of data that you want to take columns from. Check out the API reference entry on the <a href="/dms-viz-docs/preparing-data/command-line-api/#join-data"><code>--join-data</code></a> flag for more details.</p><p>The <strong>Join Data</strong> must contain a <a href="#site-or-reference-site"><code>site</code></a>, <a href="#wildtype"><code>wildtype</code></a>, and <a href="#mutant"><code>mutant</code></a> column, as these are used to join your incoming data with the <a href="#input-data">Input Data</a>.</p>',22),r=[n];function s(d,c,l,h,u,m){return t(),a("div",null,r)}const b=e(i,[["render",s]]);export{f as __pageData,b as default};
