import{_ as e,o as t,c as a,Q as i}from"./chunks/framework.9ad6f510.js";const f=JSON.parse('{"title":"Data Requirements","description":"","frontmatter":{},"headers":[],"relativePath":"preparing-data/data-requirements/index.md","filePath":"preparing-data/data-requirements/index.md"}'),o={name:"preparing-data/data-requirements/index.md"},n=i('<h1 id="data-requirements" tabindex="-1">Data Requirements <a class="header-anchor" href="#data-requirements" aria-label="Permalink to &quot;Data Requirements&quot;">​</a></h1><p>To use <strong><code>dms-viz</code></strong>, you&#39;ll need two files. First, you&#39;ll need some <a href="#input-data">input data</a> that contains the mutation-based data you&#39;d like to visualize. Second, you&#39;ll need <a href="#sitemap">a map</a> of the sites that are mutated in your dataset to the sites in the reference and protein structure.</p><p><em>optionally</em>, if you have <a href="#join-data">additional data files</a>, you can join these with your input data.</p><p>Below are the detailed requirements for each data file along with example datasets.</p><h2 id="input-data" tabindex="-1">Input Data <a class="header-anchor" href="#input-data" aria-label="Permalink to &quot;Input Data&quot;">​</a></h2><p>The <strong>Input Data</strong> is the data that you&#39;d like to summarize and visualize on an interactive protein structure. It must contain a column with a quantitative metric that&#39;s associated with mutations in a protein sequence. For example, this data could be a fitness score associated with mutations to a protein, or a score that represents how a mutation changes antibody binding to an antigen. For detailed examples of <strong><code>dms-viz</code></strong> in action, check out these <a href="/dms-viz-docs/visualizing-data/vignettes/">Vignettes</a>.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The input data must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><p>The input data must contain the following columns with <strong>exactly</strong> these names:</p><ul><li><h3 id="site-or-reference-site" tabindex="-1"><code>site</code> or <code>reference_site</code> <a class="header-anchor" href="#site-or-reference-site" aria-label="Permalink to &quot;`site` or `reference_site`&quot;">​</a></h3><p>This column should contain the <strong>site</strong> in the protein at which each measurement was made. This column can be numeric (i.e., <code>[1, 2, 3, 4]</code>) or it can contain strings (i.e., <code>[1, 2, 2a, 2b, 3]</code>). Additionally, the sites do not need to be continuous (i.e., <code>[1, 4, 5, 8]</code>). The order of your sites will be specified in the <a href="#sitemap">Sitemap</a> using the <code>sequential_site</code> column. These sites will label the x-axis of all summary plots in <strong><code>dms-viz</code></strong>.</p><p>For more details on what we mean by &#39;reference_site&#39;, check out the <a href="/dms-viz-docs/preparing-data/data-requirements/#reference-site">description of the sitemap file</a>.</p></li><li><h3 id="mutant" tabindex="-1"><code>mutant</code> <a class="header-anchor" href="#mutant" aria-label="Permalink to &quot;`mutant`&quot;">​</a></h3><p>This column should contain the identity of the <strong>mutation</strong> that each measurement is associated with. These mutations should be represented using the <a href="https://www.bioinformatics.org/sms/iupac.html" target="_blank" rel="noreferrer">IUPAC single-letter codes</a> along with symbols for stop codons and gaps (i.e., <code>R, M, P, *, -</code>). If you need to extend or shrink this alphabet, you can do so using the <a href="/dms-viz-docs/preparing-data/command-line-api/#alphabet"><code>--alphabet</code></a> flag of <code>configure-dms-viz</code>.</p></li><li><h3 id="wildtype" tabindex="-1"><code>wildtype</code> <a class="header-anchor" href="#wildtype" aria-label="Permalink to &quot;`wildtype`&quot;">​</a></h3><p>This column should contain the <strong>wildtype</strong> identity of residues at a given site in the protein. For example, if a Proline (<code>P</code>) was mutated to an Alanine (<code>A</code>) at position 120 in the protein (<code>P120A</code>), there should be a <code>P</code> in the wildtype column for every row where the value of the site column is 120. This column will also be used to check how well the protein structure you provided matches the wildtype sites in your data. Significant discrepancies can indicate that you&#39;re <code>reference</code>, <code>sequential</code>, and <code>protein</code> sites are misaligned.</p></li></ul><hr><p>In addition to these three mandatory columns, you will also need to specify a <code>metric</code> column. The identity of this column is specified with the <code>--metric</code> flag of <code>configure-dms-viz</code>, and it can have any name:</p><ul><li><h3 id="metric" tabindex="-1"><code>&lt;metric&gt;</code> <a class="header-anchor" href="#metric" aria-label="Permalink to &quot;`&lt;metric&gt;`&quot;">​</a></h3><p>This column should contain the <strong>quantitative metric</strong> that you&#39;d like to summarize and view on a protein structure. For example, this column could be called <code>fitness</code> and contain a score that reflects how individual mutations alter a protein&#39;s fitness.</p></li></ul><hr><p><em>Optionally</em>, depending on the design of your experiment, you can also include a &quot;<em>condition column</em>&quot; that specifies how your data is grouped if there are multiple measurements per mutation:</p><ul><li><h3 id="condition" tabindex="-1"><code>condition</code> <a class="header-anchor" href="#condition" aria-label="Permalink to &quot;`condition`&quot;">​</a></h3><p>This column should only be included if there are multiple measurements in the <a href="/dms-viz-docs/preparing-data/command-line-api/#metric"><code>&lt;metric&gt;</code></a> column for the same <code>site</code>/<code>mutation</code> combinations. An example of this would be if you have a measurement like an antibody&#39;s escape for multiple &#39;epitopes&#39; in an antigen. This column contains a unique identifier that&#39;s used to delineate between these measurements for each mutation. This &#39;identifier&#39; will show up in an interactive legend next to the visualization.</p></li></ul><h2 id="sitemap" tabindex="-1">Sitemap <a class="header-anchor" href="#sitemap" aria-label="Permalink to &quot;Sitemap&quot;">​</a></h2><p>The <strong>Sitemap</strong> is a tabular <code>.csv</code> file that specifies the order of the <a href="/dms-viz-docs/preparing-data/data-requirements/#site-or-reference-site"><code>site</code></a> (<code>reference_site</code>) column in your input data and, <em>optionally</em>, how the <a href="/dms-viz-docs/preparing-data/data-requirements/#site-or-reference-site"><code>site</code></a> column corresponds to the numbering in the <a href="/dms-viz-docs/preparing-data/command-line-api/#structure">protein structure</a> you provide.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The sitemap must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><ul><li><h3 id="reference-site" tabindex="-1"><code>reference_site</code> <a class="header-anchor" href="#reference-site" aria-label="Permalink to &quot;`reference_site`&quot;">​</a></h3><p>This column <strong>must</strong> correspond to the <code>site</code> or <code>reference_site</code> column in your <a href="#input-data">input data</a>. If the <a href="/dms-viz-docs/preparing-data/data-requirements/#protein-site"><code>protein_site</code></a> isn&#39;t provided, this column is also assumed to correspond to the identity of the sites in the <a href="/dms-viz-docs/preparing-data/command-line-api/#structure">protein structure</a></p><p>The <code>reference_site</code> refers to the identity of the sites that are mutated in your dataset. These sites will ultimately label the x-axis of the visualization. These &#39;<em>reference</em>&#39; sites can sometimes differ from the <code>sequential_site</code> (<a href="/dms-viz-docs/preparing-data/data-requirements/#sequential-site">described below</a>); for example, the current SARS-CoV-2 Spike protein variants have insertions and deletions that cause the widely used Wuhan-Hu-1 &#39;reference&#39; numbering to differ from the sequential, numeric order of the data.</p></li><li><h3 id="sequential-site" tabindex="-1"><code>sequential_site</code> <a class="header-anchor" href="#sequential-site" aria-label="Permalink to &quot;`sequential_site`&quot;">​</a></h3><p>This column is the sequential order of the reference sites and <strong>must be a numeric column</strong>. This will determine the order of the protein sites in the visualizations.</p></li><li><h3 id="protein-site" tabindex="-1"><code>protein_site</code> <a class="header-anchor" href="#protein-site" aria-label="Permalink to &quot;`protein_site`&quot;">​</a></h3><p><em>Optionally</em>, this column is only necessary if the <code>reference_site</code> sites are different from the sites (residue numbering) in your provided protein structure. If they are different, this column is the position in the protein structure that corresponds to the <code>reference_site</code> values in your data.</p></li><li><h3 id="chains" tabindex="-1"><code>chains</code> <a class="header-anchor" href="#chains" aria-label="Permalink to &quot;`chains`&quot;">​</a></h3></li></ul><p><em>Optionally</em>, this column is only necessary if you&#39;ve provided the <code>protein_site</code> column and there are multiple <code>reference_site</code> sites for the same value of <code>protein_site</code>. This might be the case if your data corresponds to <em>discontinuous chains</em> in the protein structure. For example, if your data is measured over two separate chains with overlapping numbering schemes. For example, Influenza HA protein structures usually have separate chains with overlapping numbering for the stalk and the head. So the reference sites 102 and 30(HA) might both correspond to the residue number 102 in the PDB file. In that case, the only way to distinguish between them on the structure is with the identity of the chain (i.e. A vs. B). This column should have chains in the same format as the chains provided to <a href="/dms-viz-docs/preparing-data/command-line-api/#included-chains"><code>--included-chains</code></a> (i.e. a space-separated string of chains: &quot;A B C D&quot;).</p><h2 id="join-data" tabindex="-1">Join Data <a class="header-anchor" href="#join-data" aria-label="Permalink to &quot;Join Data&quot;">​</a></h2><p><em>Optionally</em>, you might have some additional data that you want to combine with your <a href="#input-data">Input Data</a>. You do this so you can include columns from this <strong>Join Data</strong> in the <a href="/dms-viz-docs/preparing-data/command-line-api/#filter-cols">filters</a> or <a href="/dms-viz-docs/preparing-data/command-line-api/#tooltip-cols">tooltips</a> of your visualization. This option helps streamline that workflow.</p><div class="warning custom-block"><p class="custom-block-title">Important!</p><p>The join data must be in <code>.csv</code> format. If your data is tabular but in another format, please convert it to <code>.csv</code>.</p></div><p>You can specify more than one <code>.csv</code> file if there are multiple sources of data that you want to take columns from. Check out the API reference entry on the <a href="/dms-viz-docs/preparing-data/command-line-api/#join-data"><code>--join-data</code></a> flag for more details.</p><p>The <strong>Join Data</strong> must contain a <a href="/dms-viz-docs/preparing-data/data-requirements/#site-or-reference-site"><code>site</code></a>, <a href="#wildtype"><code>wildtype</code></a>, and <a href="#mutant"><code>mutant</code></a> column, as these are used to join your incoming data with the <a href="#input-data">Input Data</a>.</p>',25),s=[n];function r(d,c,l,h,u,m){return t(),a("div",null,s)}const g=e(o,[["render",r]]);export{f as __pageData,g as default};
